From: =?utf-8?q?Michel_D=C3=A4nzer?= <mdaenzer@redhat.com>
Date: Fri, 16 May 2025 17:38:44 +0200
Subject: kms: Extend hotplug event timeout to 3 seconds

Per https://gitlab.gnome.org/GNOME/mutter/-/issues/4101, MSI PRO MP275Q
monitors take around 2.5 seconds for an input source scan.

Origin: https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4442
Bug: https://gitlab.gnome.org/GNOME/mutter/-/issues/1607
Bug-Ubuntu: https://launchpad.net/bugs/1971434
---
 src/backends/native/meta-kms.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/backends/native/meta-kms.c b/src/backends/native/meta-kms.c
index 3b2a130..7abed89 100644
--- a/src/backends/native/meta-kms.c
+++ b/src/backends/native/meta-kms.c
@@ -379,7 +379,7 @@ on_udev_hotplug (MetaUdev    *udev,
     }
 
   ensure_hotplug_timeout_source (kms);
-  g_source_set_ready_time (kms->hotplug_timeout, now + 2 * G_USEC_PER_SEC);
+  g_source_set_ready_time (kms->hotplug_timeout, now + 3 * G_USEC_PER_SEC);
 
   g_hash_table_insert (kms->hotplug_events, g_steal_pointer (&hotplug_event),
                        NULL);
